packages/traceability/issuer-api/src/pods/certificate/commands/batch-transfer-certificates.command.ts

```diff
@@ -0,0 +1,10 @@
+import { BigNumber } from 'ethers';
+
+export class BatchTransferCertificatesCommand {
+    constructor(
+        public readonly certificateIds: number[],
+        public readonly to: string,
+        public readonly forAddress: string,
+        public readonly values?: BigNumber[]
```

Comment:
This list is related to the list of certificateIds?

1. is length ensured to be the same length as certificateIds?
2. what if some value is missing (or is undefined in the middle of the list for example)?

To be honest I don't like that API. Wouldn't it better to accept tuples of certificateId and value?

Reply:
Yes it is.

1. It's ensured on the smart contract level.
2. The smart contract will reject it.

But you are right, tuples would be better. I just blindly followed the Solidity implementation which doesn't have tuples.

Reply:
I mean, object probably instead of tuple tuple :D

Reply:
Because I can make a transfer for one certificate to transfer all the energy, and for another one only part